### PR TITLE
fix(signals): several sighandler fixes

### PR DESCRIPTION
**Pull request by antmat**
* remove assertion on default action, now default action does nothing on all signals
* fix deadlock on multiple terminate calls
* do not throw exception during sigwaitinfo on EINTR
* use wrapper with source attribute to log in signal handler thread
* initialize sigset with zero value, do not rely on sigpending call
* more verbose logging
* reduce contention - free lock before cv.notify_one

I will also make a PR to v0.12.6

### CODE REVIEW

File: src/runtime/runtime.cpp [hunk 1]
```diff
@@ -117,8 +117,10 @@ struct sigchild_handler_t {
 };
 
 void terminate() {
-    std::unique_lock<std::mutex> lock(finalizer_mutex);
-    finalize = true;
+    {
```
Comment (3Hren) [inline src/runtime/runtime.cpp]: Either use `lock.unlock` or `std::lock_guard` instead.

File: src/runtime/runtime.cpp [hunk 2]
```diff
@@ -282,7 +285,7 @@ main(int argc, char* argv[]) {
     signal_handler.async_wait(SIGTERM, terminate_handler_t());
 
     // Start signal handling thread
-    std::thread sig_thread(&run_signal_handler, std::ref(signal_handler), std::ref(*logger));
```
Comment (3Hren) [inline src/runtime/runtime.cpp]: You can just create another wrapper using `context.logger("signal_handler")`.

File: src/runtime/runtime.cpp [hunk 3]
```diff
@@ -299,7 +302,12 @@ main(int argc, char* argv[]) {
 
     // Wait until signaling termination
     std::unique_lock<std::mutex> lock(finalizer_mutex);
-    finalizer_cv.wait(lock, [&]{return finalize;});
+
+    if(!finalize) {
```
Comment (3Hren) [inline src/runtime/runtime.cpp]: Not necessary to check, CV works the following way:
```c++
while (!fn()) {
    wait(lock);
}
```

File: src/signal.cpp [hunk 4]
```diff
@@ -48,7 +48,7 @@ int wait_wrapper(sigset_t* set, siginfo_t* info) {
 }
 
 bool pending() {
-    sigset_t set;
+    sigset_t set = sigset_t();
```
Comment (3Hren) [inline src/signal.cpp]: `= {};`

File: src/signal.cpp [hunk 5]
```diff
@@ -191,15 +187,21 @@ handler_t::run() {
     sigset_t sig_set = get_signal_set(signals);
     while(should_run || pending()) {
         const int signal_num = wait_wrapper(&sig_set, &info);
-        COCAINE_LOG_DEBUG(logger, "caught signal: {}", signal_num);
         if(signal_num == -1) {
-            throw std::system_error(std::error_code(errno, std::system_category()), "sigwaitinfo failed");
+            COCAINE_LOG_WARNING(logger, "error in sigwaitinfo: {} - {}", signal_num, errno);
```
Comment (3Hren) [inline src/signal.cpp]: It will look like "error in sigwaitinfo: 23 - 42", which looks awkward.

File: src/signal.cpp [hunk 6]
```diff
@@ -191,15 +187,21 @@ handler_t::run() {
...
+                continue;
+            }
         }
         // Do not process interruption signal in case stop() was received.
         // As we use SIGCONT there is no harm of doing this
         if(signal_num == interrupt_signal && !should_run) {
+            COCAINE_LOG_INFO(logger, "skipping sighandler internal interrupt signal - {}", signal_num);
             continue;
         }
+        COCAINE_LOG_INFO(logger, "caught signal: {}", signal_num);
```
Comment (3Hren) [inline src/signal.cpp]: Do we have some kind of signum => signame mapping? If so it's better to use ut.

File: src/runtime/runtime.cpp (hunk 2 above)
Comment (antmat) [inline src/runtime/runtime.cpp]: nope, scope is different

**Pull request merged**